runtime/runtime/src/actions.rs

```diff
@@ -473,9 +476,32 @@ pub(crate) fn action_implicit_account_creation_transfer(
         }
         // Invariant: The `account_id` is implicit.
         // It holds because in the only calling site, we've checked the permissions before.
-        AccountType::EthImplicitAccount | AccountType::NamedAccount => {
-            panic!("must be near-implicit")
+        AccountType::EthImplicitAccount => {
+            if checked_feature!("stable", EthImplicit, current_protocol_version) {
+                // TODO(eth-implicit) Use real Wallet Contract.
+                let wallet_contract = wallet_contract_placeholder();
+                let storage_usage = fee_config.storage_usage_config.num_bytes_account
+                    + fee_config.storage_usage_config.num_extra_bytes_record
+                    + wallet_contract.code().len() as u64;
+
+                *account =
+                    Some(Account::new(transfer.deposit, 0, *wallet_contract.hash(), storage_usage));
+
+                set_code(state_update, account_id.clone(), &wallet_contract);
+                // Precompile the contract and store result (compiled code or error) in the database.
+                // Note, that contract compilation costs are already accounted in deploy cost using
+                // special logic in estimator (see get_runtime_config() function).
+                precompile_contract(
+                    &wallet_contract,
+                    &apply_state.config.wasm_config,
+                    apply_state.cache.as_deref(),
+                )
+                .ok();
+            } else {
+                panic!("must be near-implicit");
```

Comment:
Can this be safely merged? This code needs to be safe before and after the new protocol feature is stable. Can you add a comment explaning why this panic is unreachable?